src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -3104,13 +3106,97 @@ public override IOperation VisitObjectCreation(IObjectCreationOperation operatio
             ImmutableArray<IArgumentOperation> visitedArgs = VisitArguments(operation.Arguments);
 
             // Initializer is removed from the tree and turned into a series of statements that assign to the created instance
-            var objectCreation = new ObjectCreationExpression(operation.Constructor, initializer: null, visitedArgs, semanticModel: null, operation.Syntax, operation.Type, operation.ConstantValue, operation.IsImplicit);
+            IOperation objectCreation = new ObjectCreationExpression(operation.Constructor, initializer: null, visitedArgs, semanticModel: null, operation.Syntax, operation.Type, operation.ConstantValue, operation.IsImplicit);
+
+            if (operation.Initializer != null)
+            {
+                SpillEvalStack();
+
+                int initializerCaptureId = _availableCaptureId++;
+                AddStatement(new FlowCapture(initializerCaptureId, objectCreation.Syntax, objectCreation));
 
-            // PROTOTYPE(dataflow): For a non-null initializer, we'll need to assign the created object to a flow reference and rewrite all initializers to assign to/call functions on it, then return return the flow reference
+                objectCreation = new FlowCaptureReference(initializerCaptureId, objectCreation.Syntax, objectCreation.Type, objectCreation.ConstantValue);
```

Comment:
>objectCreation [](start = 16, length = 14)

It feels like the name ```objectCreation``` is somewhat confusing after this point. It is no longer an object creation. Consider renaming the local to "createdInstance" or something similar.